src/detect.c

```diff
@@ -373,15 +373,65 @@ static int DetectLoadSigFile(DetectEngineCtx *de_ctx, char *sig_file,
     }
     fclose(fp);
 
-    (*goodsigs) = good;
-    (*badsigs) = bad;
+    *goodsigs = good;
+    *badsigs = bad;
     return 0;
 }
 
 /**
+ *  \brief Expands wildcards and reads signatures from each matching file
+ *  \param de_ctx Pointer to the detection engine context
+ *  \param sig_file Filename (or pattern) holding signatures
+ *  \retval -1 on error
+ */
+static int ProcessSigFiles(DetectEngineCtx *de_ctx, char *pattern,
+        SigFileLoaderStat *st, int *good_sigs, int *bad_sigs)
+{
+    if (pattern == NULL) {
+        SCLogError(SC_ERR_INVALID_ARGUMENT, "opening rule file null");
+        return -1;
+    }
+
+    glob_t files;
+    int r = glob(pattern, 0, NULL, &files);
+
+    if (r == GLOB_NOMATCH) {
+        SCLogWarning(SC_ERR_NO_RULES, "No rule files match the pattern %s", pattern);
+        ++(st->bad_files);
+        return -1;
+    } else if (r != 0) {
+        SCLogError(SC_ERR_OPENING_RULE_FILE, "error expanding template %s: %s",
+                 pattern, strerror(errno));
+        return -1;
+    }
+
+    for (size_t i = 0; i < files.gl_pathc; i++) {
```

Comment:
On 05/21/2015 06:19 PM, Alexander Gozman wrote:
> Seems to be a buggy header, because gl_pathc in struct glob_t should be
> size_t. Maybe, type cast on gl_pathc will do the trick?

This was on OpenBSD 5.4. The gl_pathc member is a regular int there.